docs/solver-pipeline.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: default
+title: Solver Pipeline
+subtitle: Overview of solver pipeline and related command-line options
+slug: solver-pipeline
+---
+
+{:.toc}
+Contents
+{:.toc__title .no_toc}
+* Table of contents placeholder
+{:.toc__list .list-anchor}
+{:toc}
+
+# The solver pipeline
+
+KLEE has a number of command line options that control the solver pipeline that
+is shared both by the `klee` and `kleaver` tools.
+
+The solver pipeline is implemented internally using the decorator design pattern.
+Thus when a solver query is issued it travels through a nested stack of solvers
+and may eventually call the underlying SMT solver (core solver).
+
+The reasons for this pipe line architecture are:
+
+1. Allows transforming the solver query into another form that is hopefully
+   faster for the core solver to solve (e.g. the Independence solver).
+2. Allows querying the core solver to be avoided in some cases (e.g. the
+   Counter example caching solver).
+3. Allows logging of queries at different stages in the pipeline.
+
+The `constructSolverChain()` function is used to create the solver pipeline.
+
+## Core solver
+
+This is the underlying SMT solver. Several solvers are currently supported.
+The default is dependent on what solvers were available when KLEE was built.
+The solver can be selected using the `-solver-backend=` option.
+
+1.  **STP:** Simple Theorem Prover SMT solver [link](http://stp.github.io). Use `-solver-backend=stp`.
+2.  **Z3:** The Z3 Theorem Prover [link](https://github.com/Z3Prover/z3). Use `-solver-backend=z3`.
+3.  **metaSMT:** An Embedded Domain Specific Language for SMT [link](http://www.informatik.uni-bremen.de/agra/eng/metasmt.php). Use `-solver-backend=metasmt`.
+
+When using metaSMT as the core solver, its backend can be specified using the
+`-metasmt-backend=` option.
+
+## Other Solvers
+
+The remaining solvers rely on an underlying solver to do their work.
+
+### Assignment validating solver
+
+This solver checks that when an assignment is requested in the solver pipeline
```

Comment:
The wording is not very clear. I would change the order of the sentences: First the goal, then how it is implemented.